lib/connection.ts

```diff
@@ -242,7 +242,7 @@ export class Connection extends Entity {
 
         onClose = (context: RheaEventContext) => {
           removeListeners();
-          const err = context.error || context.connection.error;
+          const err = context.error || context.connection.error || Error('Failed to connect');
```

Comment:
@amarzavery Won't we get the `onClose` event when the connection is closed explicitly?

Reply:
Ah, never mind. All these listeners are removed as soon as the connection opens.